### DISCUSSION TITLE
Use `filePath` instead of `fileName` when running Abaqus with script

### DISCUSSION THREAD

**Discussion by diogo-rossi** [haiiliin/abqpy#1095] [Ideas]
I'm thinking about the case when we want to keep the main python script in one directory and run the Abaqus simulations in another directory. This could be the case when we want to run multiple models, each one in a separate folder, recording all the Abaqus files in that folders. This could be done using the `os.chdir` function in Python.

To achieve that, we may need to change the parameter of the Abaqus command to `filePath` instead of `fileName`, so that Abaqus can find the script to run it.

```python
os.system(f"{abaqus} cae -noGUI {filePath}")
```

Then, the method `saveAs` in the class `Mdb` should be modified to the following:
```python
    def saveAs(self, pathName: str):
        abaqus = "abaqus"
        if "ABAQUS_BAT_PATH" in os.environ.keys():
            abaqus = os.environ["ABAQUS_BAT_PATH"]

        filePath = os.path.abspath(sys.argv[0])

        try:  # If it is a jupyter notebook
            import ipynbname
            filePath = ipynbname.path()
            os.system(f"jupyter nbconvert --to python {filePath}")
            filePath = filePath.replace(".ipynb", ".py")
        except:
            pass
        os.system(f"{abaqus} cae -noGUI {filePath}")
```

If you agree with that, I can submit a PR, also modifying the method `openOdb` as well.

**Comment by haiiliin** [comment]
In the current implementation, I use the shell command `cd fileDir` to change the work directory, and then use `abaqus cae -noGUI fileName` to submit the script, I believe this is equivalent to directly use `abaqus cae -noGUI filePath` to submit the script.

**Reply by diogo-rossi** [reply]
Actually, the current shell command `os.system(f"cd {fileDir}")` is not changing the work directory (I've made some tests).
Apparently, it's behavior is the same as the `os.system('exit')`, which causes nothing. Please, see [this question](https://stackoverflow.com/questions/35277128/why-does-not-os-systemcd-mydir-work-and-we-have-to-use-os-chdirmydir-ins) and also [this other question](https://stackoverflow.com/questions/21406887/subprocess-changing-directory).

If we keep the current implementation, I think we will need to change the line `os.system(f"cd {fileDir}")` to `os.chdir(f"{fileDir}")` to work correctly. But I think this modification will add a good improvement feature (see the other comment).

**Comment by haiiliin** [comment]
Even though, directly using `filePath` seems a better choice, it is more simple.

**Comment by haiiliin** [comment]
> Even though, directly using `filePath` seems a better choice, it is more simple. 

But when we want to submit the job in a script/ if we directly use `filePath` without setting the work directly, all the temporary files will be in the folder you run you script, instead of in the folder that contains the script file, which I think it should be

**Reply by diogo-rossi** [reply]
Yes, exactly. If we use `filePath` in the command, when we run `python path/to/file/scripty.py` from a specific folder, the "work directory" will be that folder, and all the temporary files will be in that folder, but the script will be located on `path/to/file/`.

This behavior will let us submit the job in a folder different from the script's folder, and also allowing us to change the work directory inside the script, to run multiple models in multiple folders.

See the following example. Here, I change the work directory in the second line
```python
import os
os.chdir('path/to/simulation/folder') # change the directory

# Now the CWD (current work directory) is on the 'simulation/folder'

from abaqus import *
from abaqusConstants import *
from driverUtils import *

... # do stuff to build the model

mdb.saveAs('myFile.cae')
# now the line `abaqus cae -noGUI filePath` will run on the CWD, 
# but it will be able to find the path of this script
```
I could use the function `os.chdir` multiple times in the script, and run the Abaqus command in different folders.
I've made some tests (with the modified `Mdb.saveAs`) method, and it worked.

**Reply by haiiliin** [reply]
I understand, I think what you are suggesting is the "general" behavior. 

The difference between these two implementations is whether we have to explicitly set the work directory to the script's path when we run the script, the general way is not to set. 

I am doing this because in most conditions we want the work directory to be the script's path even if we run the script in another path. But this kind of behavior is inconsistent with the general behavior. So it is just for convenience, but it is not correct. 

Let's change it to the right way, and use the `filePath` to submit the script.

**Reply by diogo-rossi** [reply]
Ok, I will create a new issue to record this, and then submit a PR with the modification.
I'll also make a commit modifying the initial lines of the tests to reflect these changes (but will submit for validation)

**Reply by haiiliin** [reply]
OK, thanks.

**Comment by haiiliin** [comment]
See also #1103 and #1104.
